.github/workflows/ci.yaml

```diff
@@ -75,7 +75,7 @@ jobs:
         id: go
         uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          go-version: '1.21.1'
+          go-version: '1.21.3'
```

Comment:
can we use just `'1.21'` with the check-latest it will always pick the up-to-date go

Reply:
and will help us reducing the things to update :)

Reply:
maybe :) I did left 1.21.3 to intentionally map what was the version, as we also use this same "version" on go.mod. (And also was easier to me to do a grep and see if I were missing somewhere! :P ) 

But yeah, let me do it as 1.21 at least on CI

Do you think we should do it on other parts of code changed as well?

Reply:
On the other hand, using `1.21.3` explicitly makes it obvious that it runs with patched version, while using just 1.21 make it a bit more ambiguous.

Reply:
sounds good to me, i was trying to make less changes to us